### PR TITLE
Add Forms

### PR DESCRIPTION
**Pull request by Ibizugbe**
In this PR, I have completed the current milestone of the project and kept to the given requirements.
- no linter errors.
- the project is professionally documented.
- the right git-flow method has been used.
- A method called `current_user` has been created in the `ApplicationController` to make current user data available to all controllers.
- Users can now like and comment.
- A form has been created to create a post.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -5,6 +5,37 @@ def index
...
+    @user = User.find(params[:user_id])
+    @post = Post.new
+  end
+
+  def create
+    # @user = User.find(params[:user_id])
+    @post = Post.new(post_params)
+    @post.author = current_user
+    @post.comment_counter = 0
+    @post.like_counter = 0
```
Comment (codepantha) [inline app/controllers/posts_controller.rb:22]: - [ ] Please remove the commented line of code above as it doesn't make your code look clean 👍 

- [ ] [OPTIONAL] Please try to set a default value of 0 for the `comments_counter` and `likes_counter` at the database level so you don't have to worry about it in your code like you have done here 👍

File: app/controllers/likes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class LikesController < ApplicationController
+  def create
+    post = Post.find(params[:post_id])
+    like = Like.new
+    like.author = current_user
+    like.post = post
```
Comment (codepantha) [inline app/controllers/likes_controller.rb:6]: - [ ] [OPTIONAL] You have implemented the like functionality correctly, however I would like to show you a way you can make use of the relationships between the user and the likes to make your code even shorter and more readable.
Remember that there is a one to many relationship between a user and the likes, right? Now you can do something like this 👇 
``` ruby
like = current_user.likes.new(post_id: post.id)
like.save
```

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRES CHANGES ♻️ 

## Hi @Ibizugbe 👋, 

You have done an excellent job on this project so far 👏. There is only one very little change that is required. Please note that I have also suggested some optional changes as well which would definitely make your code better if you apply them.



## Highlights 🎯 
✔️ Descriptive PR
✔️ No linter errors
✔️ Used correct GitFlow

 ## Required Changes ♻️
_Check the comments under the review._

- [ ] Please add an MIT license to your project and link it in your README. It is important that every software (open source ones, too) should have a license. Currently, the license link is broken.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Ibizugbe) [reply app/controllers/posts_controller.rb:22]: Thanks a lot for this suggestion, I will do well to implement them👍.

File: app/controllers/likes_controller.rb (hunk 2 above)
Comment (Ibizugbe) [reply app/controllers/likes_controller.rb:6]: Thanks, @codepantha for these suggestions. I will surely adopt these changes👍

**Comment by Ibizugbe** [comment]
Thanks a lot, @codepantha I have fixed the license and I'll ensure I adopt that practice in future repositories.

**Review by danielufeli — APPROVED** [review]
Hi @Ibizugbe ,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

a

## Highlights
- Nice commits messages✔️
- Descriptive PR ✔️
- Good use of GitHub flow ✔️
- Clean Code✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @danielufeli in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Ibizugbe** [comment]
Thanks @danielufeli  for reviewing and approving this PR.

**Pull request merged**